pkg/workspace/library_loader.go

```diff
@@ -0,0 +1,182 @@
+	OutputFiles []files.OutputFile
+}
+
+// EvalValuesAst represents in AST form the data.values to be fed to the
+// evaluation process.
+type EvalValuesAst interface{}
+
+func NewLibraryLoader(lib *Library, ui files.UI, loaderOpts TemplateLoaderOpts) *LibraryLoader {
+	return &LibraryLoader{
+		UI:      ui,
+		Opts:    loaderOpts,
+		Library: lib,
+	}
+}
+
+func (ll *LibraryLoader) LoadValues(parentValues EvalValuesAst) (EvalValuesAst, error) {
+	valuesFiles, err := ll.loadValueFiles(ll.Library)
+	if err != nil {
+		return nil, err
+	}
+
+	noValuesLoader := NewTemplateLoader(nil, ll.UI, ll.Opts)
+
+	vals, err := DataValuesPreProcessing{valuesFiles, parentValues, noValuesLoader, ll.Opts.IgnoreUnknownComments}.Apply()
+	if err != nil {
+		return nil, err
+	}
+
+	return vals, nil
+}
+
+func (ll *LibraryLoader) loadValueFiles(lib *Library) ([]*FileInLibrary, error) {
+	var valuesFiles []*FileInLibrary
+
+	for _, fileInLib := range lib.ListAccessibleFiles() {
+		if fileInLib.File.Type() == files.TypeYAML && fileInLib.File.IsTemplate() {
+			fileBs, err := fileInLib.File.Bytes()
+			if err != nil {
+				return nil, err
+			}
+
+			docSet, err := yamlmeta.NewDocumentSetFromBytes(fileBs, fileInLib.File.RelativePath())
+			if err != nil {
+				return nil, fmt.Errorf("Unmarshaling YAML template '%s': %s", fileInLib.File.RelativePath(), err)
+			}
+
+			tplOpts := yamltemplate.MetasOpts{IgnoreUnknown: ll.Opts.IgnoreUnknownComments}
+
+			valuesDocs, _, err := yttlibrary.DataValues{docSet, tplOpts}.Extract()
+			if err != nil {
+				return nil, err
+			}
+
+			if len(valuesDocs) > 0 {
+				valuesFiles = append(valuesFiles, fileInLib)
+				fileInLib.File.MarkForOutput(false)
+			}
+		}
+	}
+
+	return valuesFiles, nil
+}
+
+func (ll *LibraryLoader) Eval(values EvalValuesAst) (*EvalResult, error) {
+	res := &EvalResult{
+		DocSet:      &yamlmeta.DocumentSet{},
+		OutputFiles: []files.OutputFile{},
+	}
+	overlayProcess := &OverlayPostProcessing{
+		docSets: map[*FileInLibrary]*yamlmeta.DocumentSet{},
+	}
+
+	err := ll.evalLibrary(ll.Library, "", values, res, overlayProcess)
+	if err != nil {
+		return nil, err
+	}
+
+	res.DocSets, err = overlayProcess.Apply()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fileInLib := range sortedOutputDocSets(res.DocSets) {
+		docSet := res.DocSets[fileInLib]
+		res.DocSet.Items = append(res.DocSet.Items, docSet.Items...)
+
+		resultDocBytes, err := docSet.AsBytes()
+		if err != nil {
+			return nil, fmt.Errorf("Marshaling template result: %s", err)
+		}
+
+		ll.UI.Debugf("### %s result\n%s", fileInLib.Path, resultDocBytes)
+		outputFile := files.NewOutputFile(fileInLib.Path, resultDocBytes)
+		res.OutputFiles = append(res.OutputFiles, outputFile)
+	}
+
+	return res, nil
+}
+
+func (ll *LibraryLoader) evalLibrary(lib *Library, libPath string, values EvalValuesAst, res *EvalResult, overlayProcess *OverlayPostProcessing) error {
+	loader := NewTemplateLoader(values, ll.UI, ll.Opts)
+
+	for _, file := range lib.files {
```

Comment:
Use `ListAccessibleFiles` instead

Reply:
Except, relative paths needs to be constructed, so unless it's constructed in ListAccessibleFiles, we'll need to keep the current form